app/jobs/webhook_deliveries/deliver_job.rb

```diff
@@ -0,0 +1,40 @@
+module WebhookDeliveries
+  class DeliverJob < ApplicationJob
+    queue_as :default
+
+    def perform(event_type, course, resource)
+      webhook_endpoint = course.webhook_endpoint
+      payload = {
+        data: data(event_type, resource),
+        event: event_type
+      }
+      uri = URI.parse(webhook_endpoint.webhook_url)
+      request = Net::HTTP::Post.new(uri.request_uri)
+      request['Content-Type'] = 'application/json'
+      request.body = payload.to_json
+      http = Net::HTTP.new(uri.host, uri.port)
+      http.use_ssl = (uri.scheme == 'https')
+      response = http.request(request)
```

Comment:
Could we enforce a strict timeout here? A configurable value, stored as milliseconds, read from an environment variable?